app/assets/javascripts/discourse/tests/acceptance/sidebar-narrow-desktop-test.js

```diff
@@ -0,0 +1,53 @@
+import { test } from "qunit";
+
+import { click, settled, visit, waitUntil } from "@ember/test-helpers";
+
+import { acceptance, exists } from "discourse/tests/helpers/qunit-helpers";
+
+acceptance("Sidebar - Narrow Desktop", function (needs) {
+  needs.user();
+
+  needs.settings({
+    enable_experimental_sidebar_hamburger: true,
+    enable_sidebar: true,
+  });
+
+  needs.hooks.beforeEach(function () {
+    window.innerWidth = 1280;
+  });
+  needs.hooks.afterEach(function () {
+    $("body").width(1280);
+  });
```

Comment:
O sorry this is not what I meant and we definitely don't want to be altering the width. I think in our tests, we should just set `site.narrowDesktopView` directly and ensure the application responds correctly to the changes.

Reply:
I think I made it not persistently green. Do you still think that it is a better approach to modify site.narrowDesktopView? I feel that changing width is testing behaviour when you resize window. So we are making sure that ResizeObserver works

Reply:
I think the test will end up being flaky... but I'm unsure how flaky it will be. The other concern is that if I'm running the tests locally, having the window width change might be surprising/annoying. Since we've already written the code, I think we can just try it out and only change the tests if we encounter problems?